### PR TITLE
clean up ML_LOG(FATAL) and MG_CHECK

### LINKED ISSUE #717: MG_LOG(FATAL) doesn't print a callstack

### PR DESCRIPTION
**Pull request by tommadams**
Fixes #717

### CODE REVIEW

File: cc/logging.cc [hunk 1]
```diff
@@ -24,15 +24,40 @@
...
+  if (level == LogLevel::INFO) {
+    // We don't add a prefix to MG_LOG(INFO) log lines because many things rely
+    // on the exact string being printed (GTP, correct formatting of position &
+    // node descriptions, etc).
+    return;
+  }
+
   char c;
   switch (level) {
     case LogLevel::INFO:
```
Comment (sethtroisi) [inline cc/logging.cc]: should this case be deleted?

File: cc/logging.cc [hunk 2]
```diff
@@ -24,15 +24,40 @@
...
 namespace internal {
+
 namespace {
+
 absl::Mutex* mutex() {
   static auto* m = new absl::Mutex();
   return m;
 }
+
+void DumpStackTrace(std::ostream* os) {
```
Comment (sethtroisi) [inline cc/logging.cc]: potentially stupid question: can you pass std::ostream& os and avoid (*os)?

Comment (tommadams) [reply cc/logging.cc]: I could, but we try and follow the google style guide, which only allows for const reference parameter.

File: cc/logging.cc (hunk 1 above)
Comment (tommadams) [reply cc/logging.cc]: good catch, thanks!

**Pull request merged**